### PR TITLE
(maint) Update supported platforms

### PR DESCRIPTION
**Pull request by MikaelSmith**
Updates supported platforms for new package releases, AIX PowerPC 8
hardware, and platforms that were added but not noted as supported.

Note that specific upgrade scenarios will still depend on Puppet being
installed from packages prior to upgrade. `puppet-agent` packages should
exist for all supported platforms.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -44,8 +44,6 @@
     } elsif $::operatingsystem == 'aix' and $::architecture =~ /PowerPC_POWER[5,6,7]/ {
       $aix_ver_number = regsubst($::platform_tag,'aix-(\d+\.\d+)-power','\1')
       $_package_file_name = "${puppet_agent::package_name}-${puppet_agent::params::master_agent_version}-1.aix${aix_ver_number}.ppc.rpm"
-    } elsif $::operatingsystem == 'Darwin' and $::macosx_productversion_major =~ /10\.[9,10,11]/ {
```
Comment (MikaelSmith) [inline manifests/init.pp]: This section was redundant, see lines 42-43 above.

Comment (highb) [inline manifests/init.pp]: Hm, must have been a bad re-base.

File: manifests/params.pp [hunk 2]
```diff
@@ -67,6 +67,12 @@
     }
   }
 
+  # Treat Amazon Linux just like Enterprise Linux 6
+  $pe_repo_dir = ($::operatingsystem == 'Amazon') ? {
```
Comment (highb) [inline manifests/params.pp]: :+1: to keeping the platform variables contained to `params.pp`

**Comment by highb** [comment]
:+1: sane changes to me, travis is passing. We just need to ensure that we test the module out on AWS AMI images at some point.

File: README.markdown [hunk 3]
```diff
@@ -116,6 +112,7 @@ Specifically in the 1.1.0 Release:
 * For Windows, you must trigger an agent run after upgrading so that Puppet can create the necessary directory structures.
 * Upgrading from 2015.2.x to 2015.3.x is not yet supported.
 * Solaris 11 is not yet supported.
+* AIX package names are based on PowerPC architecture version. PowerPC 8 is not yet supported.
```
Comment (MikaelSmith) [inline README.markdown]: We could fix this by dropping the exclusion of PowerPC 8. Happy to do that here or another PR.

**Comment by MikaelSmith** [comment]
I tested the Amazon AMI manually for OSS. It's a challenge to setup, they don't provide `ruby(selinux)`, and the packages assume Ruby 1.8, so you have to mess with the environment a lot and manually install from RPMs. It also won't upgrade from Puppet3/Facter2, which are the Amazon-provided packages. So it doesn't really work for OSS, but that's not new.

I expect PE works, but that requires setting up a PE master in AWS to test against.

File: README.markdown (hunk 3 above)
Comment (highb) [inline README.markdown]: Yeah, either is fine to me. Maybe lets get it in another PR so we don't overload this one.

**Comment by highb** [comment]
:+1: Let's just make another PR for the Power8 change.
